### ISSUE TITLE
@Mapping with defaultValue and method argument where is missing source does nothing

### ISSUE THREAD

**Issue by Saljack** [mapstruct/mapstruct#2220]
If you have an `@Mapping` with `defaultValue` and all method argument doesn't contains the source (by name of the target), MapStruct does nothing it simply ignores it (it fails in 1.4.0.Final but is not the root cause see #2214 ). You should use `constant` if there is no source. I think it will be better to fail during MapStruct processing because it is really easy to overlook it. Another solution of this problem can be use the `defaultValue` as `constant`.
```java
@Mapper
public interface MapperWithoutSource {

  class Tenant {
    public String id;
    public String type;
  }

  class TenantDTO {
    public String id;
  }
  
  /**
   *  There is no source "type" in TenantDTO so returned object has type = null. 
   *  I would expect failing during MapStruct processing or type = "STANDARD"
   */
  @Mapping(target="type", defaultValue = "STANDARD")
  Tenant map(TenantDTO tenant);

}
```

**Comment by filiphr** [comment]
Thanks for the report @Saljack, this is actually the same problem as in #2214. When that regression is fixed there will be a compile error saying that there is no `type` in the source `TenantDTO`. 

I think that doing a fallback to constant is error prone and complex in this case. Having a compile error is enough and the user can fix that immediately in the annotation.

**Comment by Saljack** [comment]
Just for clarification there is no error message in 1.3.1.Final too so I think it isn't the same as #2214 , that was the reason why I created this separete issue.

**Comment by filiphr** [comment]
OK not having an error message is wrong then. We will fix that.

Thanks for opening the issue.

**Comment by filiphr** [comment]
I was looking at this now and I realised that indeed it fails on 1.4.0.Final with the error message:

>The type of parameter "tenant" has no property named "type". Please define the source property explicitly.

This error message is correct, because when you are using `@Mapping(target = "type", defaultValue = "STANDARD")` it means that you want to defined the default value for the target. When `ignored`, `constant` or `javaExpression` are missing then `source` is implied. In this case the mapping will not be possible because there is no getter for `type` in the source.

Basically it is identical to having written `@Mapping(target = "type", source = "type", defaultValue = "STANDARD")`.

**Comment by myucxy** [comment]
我感觉应该是 在没有写source 时, target 的 value = defaultvalue;
当写source时, source的getter = null时, target 的 value = defaultvalue;
